spotbugs-annotations/src/main/java/edu/umd/cs/findbugs/annotations/SuppressMatchType.java

```diff
@@ -0,0 +1,22 @@
+package edu.umd.cs.findbugs.annotations;
+
+public enum SuppressMatchType {
+
+    /**
+     * Suppress bugs matching any of:
+     * <li/> the given bug type {@link String#startsWith(String)}
+     * <li/> the given bug category {@link String#equalsIgnoreCase(String)}
+     * <li/> the given bug abbreviation {@link String#equalsIgnoreCase(String)}
+     */
+    DEFAULT,
+
+    /**
+     * Suppress bugs matching the exact (case sensitive) given type.
+     */
+    EXACT,
+
+    /**
+     * Suppress bug whose type match the given regular expression.
```

Comment:
IMO it's rather connected to the `value` field that to which properties is it connected to (type, category and abbreviation), so even if it's regex or exact match, both the type, the category and the abbreviation could be considered. Or if not, the comment at the `SuppressFBWarnings.value` to should show this change.

Reply:
I guess what we're aiming for here is the least surprise principle, so I _think_ it's going to be less surprising if all matchTypes use the same criteria (type, category, abbreviation)
I've changed the code to match either of the 3 for all match types